react/src/utils/fileUtils.ts

```diff
@@ -0,0 +1,21 @@
+import { FileData } from 'chonky';
+import { File } from '../types';
+
+export const serializeToChonkyFile = (
+  file: File,
+  allowedFileExtensions: string[]
+): FileData => ({
+  id: file.path,
+  name: file.name,
+  size: file.size,
+  modDate: file.lastModified,
+  isDir: file.type === 'dir',
+  ext:
+    file.type !== 'dir' && file.name.lastIndexOf('.') > 0
+      ? file.name.substring(file.name.lastIndexOf('.') + 1)
+      : undefined,
+  icon: file.type === 'dir' ? 'folder' : 'file',
+  selectable:
```

Comment:
'selectable' is set to false for the files in CreateMapModal, but it's kinda hard to tell because of the styling. Maybe we can tweak that to make it clearer? Or in the short term, we could try using `isHidden` or just filter them out ourselves.

Reply:
Added task [WG-417](https://tacc-main.atlassian.net/browse/WG-417) as a follow on for this